pkg/util/k8s/client.go

```diff
@@ -42,38 +43,43 @@ const (
 
 // CreateClients creates kube clients from the given config.
 func CreateClients(config componentbaseconfig.ClientConnectionConfiguration, kubeAPIServerOverride string) (
```

Comment:
not directly related to this PR, but I think we should consider cleaning this up a little bit given the number of clients we now support, maybe by taking some inspiration from the K8s clientset:

```go
type Interface interface {
        K8s() clientset.Interface
        KubeAggregator() aggregatorclientset.Interface
        AntreaCRDs() crdclientset.Interface
        APIExtensions() apiextensionclientset.Interface
        AntreaMulticluster() mcclientset.Interface
        NetworkPolicy() policyclient.Interface
}
```

We could return an object implementing this interface in this function, and let consumers choose which client(s) they want to use.

cc @tnqn for his opinion

Reply:
I can follow up with a PR

Reply:
The suggestion looks perfect to me.